static/js/reducers/ui.js

```diff
@@ -124,10 +135,29 @@ export const INITIAL_UI_STATE: UIState = {
   couponNotificationVisibility:        false,
   navDrawerOpen:                       false,
   learnerChipVisibility:               null,
+  dialogVisibility:                    INITIAL_DIALOG_VISIBILITY_STATE
 };
 
 export const ui = (state: UIState = INITIAL_UI_STATE, action: Action) => {
   switch (action.type) {
+
+  case SHOW_DIALOG:
+    return {
+      ...state,
+      dialogVisibility: {
+        ...state.dialogVisibility,
+        [action.payload]: true
+      },
+    };
+  case HIDE_DIALOG:
+    return {
+      ...state,
+      dialogVisibility: {
+        ...state.dialogVisibility,
+        [action.payload]: false
+      },
+    };
```

Comment:
yeah, that's the plan. i'm going to create a separate issue to do some of that work